File: src/main/java/com/enzo/testemuralis/dto/ClienteResponseDTO.java
package com.enzo.testemuralis.dto;

import com.enzo.testemuralis.models.Cliente;

public record ClienteResponseDTO(
    Long id,
    String nome,
    String dataCadastro,
    ContatoResponseDTO contato,
    EnderecoResponseDTO endereco
) {
    public ClienteResponseDTO(
        Cliente cliente
    ) {
        this(
            cliente.getId(),
            cliente.getNome(),
            cliente.getDataCadastro(),
            new ContatoResponseDTO(cliente.getContato()),
            new EnderecoResponseDTO(cliente.getEndereco())
        );
    }
} 

File: dotnet/TesteMuralis/Dto/ClienteResponseDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TesteMuralis.Models;

namespace TesteMuralis.Dto
{
    public class ClienteResponseDTO
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string DataCadastro { get; set; }
        public ContatoResponseDTO Contato { get; set; }
        public EnderecoResponseDTO Endereco { get; set; }

        public ClienteResponseDTO(Cliente cliente) {
            Id = cliente.Id;
            Nome = cliente.Nome;
            DataCadastro = cliente.DataCadastro;
            Contato = new ContatoResponseDTO(cliente.Contato);
            Endereco = new EnderecoResponseDTO(cliente.Endereco);
        }
    }
}